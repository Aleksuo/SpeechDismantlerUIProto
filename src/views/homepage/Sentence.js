import React from 'react'
import { Typography, Divider } from '@material-ui/core'
import PropTypes from 'prop-types'

import { millisecondsToTimeString } from './../../utils/GeneralUtils.js'

/**
 * A stateless component for displaying the recognized sentences in the transcript
 * @param {*} props
 * @param {} ref
 * @author Aleksi Suoranta
 */
const Sentence = React.forwardRef((props, ref) => {
    const { sentence, onClick, isCurrent, wordColor } = props
    const startTime = sentence.startTime
    const style = isCurrent ? { borderColor: "#2196f3", borderStyle: "solid" } : {}
    const items = sentence.words.map((word, idx) => {

        var wordWithElapsedTime = {
            word: word.word,
            startTime: word.startTime,
            endTime: word.endTime,
            sentenceStartTime: startTime
        }

        var color = wordColor.GetColor(wordWithElapsedTime)
        return <span key={idx} style={{ color: color }}>{word.word} </span>
    })
    const refElement = isCurrent ? <div ref={ref}></div> : <div></div>
    return (
        <div>
            {refElement}
            <div onClick={onClick} style={style}>
                <Typography align="center" color="primary">{millisecondsToTimeString(sentence.startTime)}</Typography>
                <Typography paragraph={true} align="center">{items}</Typography>
                <Divider variant="middle" light={true} />
            </div>
        </div>)
})

Sentence.propTypes = {
    sentence: PropTypes.object,
    onClick: PropTypes.func,
    isCurrent: PropTypes.bool
}

export default Sentence