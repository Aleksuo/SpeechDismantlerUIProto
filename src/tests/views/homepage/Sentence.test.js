/* eslint-disable */
import React from "react"
import Enzyme, {shallow, mount,render} from "enzyme"
import Sentence from "../../../views/homepage/Sentence"
import Adapter from "enzyme-adapter-react-16"

Enzyme.configure({adapter: new Adapter()})

const emptySentence = {
    startTime: 0,
    words: []
}
describe("Sentence component", () =>{
    test("renders", () =>{ 
        const wrapper = shallow(<Sentence sentence={emptySentence}/>) 
        expect(wrapper).toMatchSnapshot()
    })
})