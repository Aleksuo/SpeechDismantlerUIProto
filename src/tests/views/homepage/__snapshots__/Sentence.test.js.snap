// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sentence component renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef
    sentence={
      Object {
        "startTime": 0,
        "words": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div />,
        <div
          style={Object {}}
        >
          <WithStyles(Typography)
            align="center"
            color="primary"
          >
            00:00
          </WithStyles(Typography)>
          <WithStyles(Typography)
            align="center"
            paragraph={true}
          />
          <WithStyles(Divider)
            light={true}
            variant="middle"
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              align="center"
              color="primary"
            >
              00:00
            </WithStyles(Typography)>,
            <WithStyles(Typography)
              align="center"
              paragraph={true}
            />,
            <WithStyles(Divider)
              light={true}
              variant="middle"
            />,
          ],
          "onClick": undefined,
          "style": Object {},
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "align": "center",
              "children": "00:00",
              "color": "primary",
            },
            "ref": null,
            "rendered": "00:00",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "align": "center",
              "children": Array [],
              "paragraph": true,
            },
            "ref": null,
            "rendered": Array [],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "light": true,
              "variant": "middle",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div />,
          <div
            style={Object {}}
          >
            <WithStyles(Typography)
              align="center"
              color="primary"
            >
              00:00
            </WithStyles(Typography)>
            <WithStyles(Typography)
              align="center"
              paragraph={true}
            />
            <WithStyles(Divider)
              light={true}
              variant="middle"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WithStyles(Typography)
                align="center"
                color="primary"
              >
                00:00
              </WithStyles(Typography)>,
              <WithStyles(Typography)
                align="center"
                paragraph={true}
              />,
              <WithStyles(Divider)
                light={true}
                variant="middle"
              />,
            ],
            "onClick": undefined,
            "style": Object {},
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "align": "center",
                "children": "00:00",
                "color": "primary",
              },
              "ref": null,
              "rendered": "00:00",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "align": "center",
                "children": Array [],
                "paragraph": true,
              },
              "ref": null,
              "rendered": Array [],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "light": true,
                "variant": "middle",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
